.github/workflows/ci-dgraph-code-coverage.yml

```diff
@@ -0,0 +1,74 @@
+name: ci-dgraph-code-coverage
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - main
+  schedule:
+    - cron: "30 0 * * *"
+jobs:
+  dgraph-code-coverage:
+    # TODO: uncomment this after release work
+    # if: github.event.pull_request.draft == false
+    runs-on: [self-hosted, x64]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get Go Version
+        run: |
+          #!/bin/bash
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; })
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Install protobuf-compiler
+        run: sudo apt-get install -y protobuf-compiler
+      - name: Check protobuf
+        run: |
+          cd ./protos
+          go mod tidy
+          make regenerate
+          git diff --exit-code -- .
+      - name: Make Linux Build and Docker Image with coverage enabled
+        run: make coverage-docker-image
+      - name: Build Test Binary
+        run: |
+          #!/bin/bash
+          # build the test binary
+          cd t; go build .
+      - name: Clean Up Environment
+        run: |
+          #!/bin/bash
+          # clean cache
+          go clean -testcache
+          # clean up docker containers before test execution
+          cd t; ./t -r
+      - name: Run Unit Tests
+        run: |
+          #!/bin/bash
+          # go env settings
+          export GOPATH=~/go
+          # move the binary
+          cp dgraph/dgraph ~/go/bin/dgraph
+          # run the tests
+          cd t; ./t --coverage=true --suite=ldbc
```

Comment:
suite=unit,ldbc as we discuused

Reply:
Yes, I am testing few items hence the PR is a mess. Will update these once done with testing.